scipy/signal/_fir_filter_design.py

```diff
@@ -834,6 +835,296 @@ def remez(numtaps, bands, desired, *, weight=None, type='bandpass',
+    ...     ax.set_ylabel('Gain (dB)')
+    ...     ax.set_title(title)
+
+    >>> taps = remez(numtaps, bands, desired, weight=weight)
+    >>> w, h = freqz(taps, worN=2000, fs=fs)
+    >>> plot_response(w, h, "Remezord Low-pass Filter")
+    >>> plt.show()
+
+    Designing a band-pass filter with the following specifications:
+        * 2-5 kHz passband frequency
+        * 22 kHz sampling frequency
+        * 1e-1 Max ripples
+
+    >>> fs = 22000  # Sampling frequency
+    >>> freqs = [1800, 2000, 5000, 5200]  # Band edges
+    >>> amps = [0, 1, 0]  # Amplitudes
+    >>> rips = [0.1, 0.1, 0.1]  # Max ripples
+    >>> numtaps, bands, desired, weight = remezord(freqs, amps, rips, fs=fs)
+    >>> taps = remez(numtaps, bands, desired, weight=weight)
+    >>> w, h = freqz(taps, worN=2000, fs=fs)
+    >>> plot_response(w, h, "Remezord Band-pass Filter")
+    >>> plt.show()
+
+    """
+    # Make sure the parameters are floating point numpy arrays:
+    freqs = np.asarray(freqs, "d")
+    amps = np.asarray(amps, "d")
+    rips = np.asarray(rips, "d")
+
+    # Validate inputs shapes/types
+    if len(amps) != len(rips):
+        raise ValueError("Number of amplitudes must equal number of ripples.")
+    if len(freqs) != 2 * (len(amps) - 1):
+        raise ValueError("Number of band edges must equal "
+                         "2*((number of amplitudes)-1)")
+    if alg == "herrmann":
+        remlplen = _remlplen_herrmann
+    elif alg == "kaiser":
+        remlplen = _remlplen_kaiser
+    elif alg == "ichige":
+        remlplen = _remlplen_ichige
+    else:
+        raise ValueError("Unknown filter length approximation algorithm.")
+
+    # Scale ripples with respect to band amplitudes:
+    rips /= amps + (amps == 0.0)
+    # Normalize input frequencies with respect to sampling frequency:
+    freqs /= fs
+
+    # Validate inputs values
+    if np.any(freqs > 0.5):
+        raise ValueError("Frequency band edges must not exceed the Nyquist "
+                         "frequency.")
+    if np.any(freqs < 0.0):
+        raise ValueError("Frequency band edges must be nonnegative.")
+    if np.any(rips < 0.0):
+        raise ValueError("Ripples must be nonnegative.")
+
+    # Find the longest filter length needed to implement any of the
+    # low-pass or high-pass filters with the specified edges:
+    f1 = freqs[0:-1:2]
+    f2 = freqs[1::2]
+    L = 0
+    for i in range(len(amps) - 1):
+        L = max(
+            (
+                L,
+                remlplen(f1[i], f2[i], rips[i], rips[i + 1]),
+                remlplen(0.5 - f2[i], 0.5 - f1[i], rips[i + 1], rips[i]),
+            )
+        )
+
+    # Cap the sequence of band edges with the limits of the digital frequency
+    # range:
+    bands = np.hstack((0.0, freqs, 0.5))
+
+    # The filter design weights correspond to the ratios between the maximum
+    # ripple and all of the other ripples:
+    weight = max(rips) / rips
+
+    return [L, bands, amps, weight]
```

Comment:
Is there a reason for returning a list over the more conventional tuple?
```suggestion
    return L, bands, amps, weight
```